### PR TITLE
Use ES6 IdentifierStart and IdentifierPart.

### LINKED ISSUE #1232: ES6 Updated identifier name
Relevant lexical grammar:

```
IdentifierName ::
    IdentifierStart
    IdentifierName IdentifierPart

IdentifierStart ::
    UnicodeIDStart
    $
    _
    \ UnicodeEscapeSequence


IdentifierPart ::
    UnicodeIDContinue
    $
    _
    \ UnicodeEscapeSequence
    <ZWNJ>
    <ZWJ>
```

where

* UnicodeIDStart: any Unicode code point with the Unicode property `ID_Start` or `Other_ID_Start`
* UnicodeIDContinue: any Unicode code point with the Unicode property `ID_Continue`, `Other_ID_Continue`, or `Other_ID_Start`

### PR DESCRIPTION
**Pull request by ariya**
In particular, this is now using Unicode derived properties:
* ID_Start and Other_ID_Start
* ID_Continue, Other_ID_Continue, Other_ID_Start

Update the tool to generate the right regular expressions for IdentifierPart
and IdentifierStart based on the appopriate Unicode properties.

Fixes #1232

### CODE REVIEW

File: tools/generate-identifier-regex.js [hunk 1]
```diff
@@ -10,42 +10,29 @@ var get = function(what) {
...
-var Pc = get('categories/Pc');
-
-var generateES5Regex = function() { // ES 5.1
-    // http://mathiasbynens.be/notes/javascript-identifiers#valid-identifier-names
-    var identifierStart = regenerate('$', '_')
-        .add(Lu, Ll, Lt, Lm, Lo, Nl)
-        .removeRange(0x010000, 0x10FFFF) // remove astral symbols
+var generateES6Regex = function() { // ES 6
+    // https://mathiasbynens.be/notes/javascript-identifiers-es6
+    var identifierStart = regenerate(get('properties/ID_Start'))
```
Comment (mathiasbynens) [inline tools/generate-identifier-regex.js]: Maybe add `.add('$', '_')` here for clarity? They’ll get removed anyway but it just makes the code match the spec more closely.

File: tools/generate-identifier-regex.js [hunk 2]
```diff
@@ -10,42 +10,29 @@ var get = function(what) {
...
         .removeRange(0x0, 0x7F); // remove ASCII symbols (Esprima-specific)
-    var identifierPart = identifierStart.clone()
-        .add('\u200C', '\u200D', Mn, Mc, Nd, Pc)
-        .removeRange(0x010000, 0x10FFFF) // remove astral symbols
+    var identifierPart = regenerate(get('properties/ID_Continue'))
+        .add(get('properties/Other_ID_Start'))
+        .add('\u200C', '\u200D')
```
Comment (mathiasbynens) [inline tools/generate-identifier-regex.js]: Same here with `$` and `_`. Just a personal preference, though.

**Comment by mathiasbynens** [comment]
LGTM!

File: tools/generate-identifier-regex.js (hunk 1 above)
Comment (ariya) [inline tools/generate-identifier-regex.js]: Good idea! I'll amend the commit to add those characters.

**Pull request closed without merging**